src/Service/CustomerInteractionService.php

```diff
@@ -0,0 +1,190 @@
+<?php
+/**
+ * Service for business logic in the customer interaction bundle.
+ *
+ * @author Conduction.nl <info@conduction.nl>, Robert Zondervan <robert@conduction.nl>
+ *
+ * @license EUPL-1.2 https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+namespace CommonGateway\CustomerInteractionBundle\Service;
+
+use App\Entity\Gateway as Source;
+use CommonGateway\CoreBundle\Service\CallService;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+class CustomerInteractionService
+{
+
+    /**
+     * The action configuration.
+     *
+     * @var array
+     */
+    private array $configuration;
+
+    /**
+     * The call service.
+     *
+     * @var CallService
+     */
+    private CallService $callService;
+
+    /**
+     * The entity manager.
+     *
+     * @var EntityManagerInterface
+     */
+    private EntityManagerInterface $entityManager;
+
+
+    public function __construct(
+        CallService $callService,
+        EntityManagerInterface $entityManager
+    ) {
+        $this->callService   = $callService;
+        $this->entityManager = $entityManager;
+
+    }//end __construct()
+
+
+    /**
+     * Find a source for a given url.
+     *
+     * @param string $url      The url of the object to be found.
+     * @param string $endpoint The resulting endpoint (the remainder of the path)
+     *
+     * @return Source|null The resulting source.
+     *
+     * @throws \Safe\Exceptions\UrlException
+     */
+    private function getSource(string $url, ?string &$endpoint): ?Source
+    {
+        // 1. Get the domain from the url
+        $parse    = \Safe\parse_url($url);
+        $location = $parse['scheme'].'://'.$parse['host'];
+
+        // 2.c Try to establish a source for the domain
+        $source = $this->entityManager->getRepository('App:Gateway')->findOneBy(['location' => $location]);
```

Comment:
You could maybe use (/improve) CoreBundle->GatewayResourceService->findSourcesForUrl() function here?
(and also remove/replace foreach below)

Reply:
It is a copy of the first part of that function. It is an idea to update this, but the follow up is really different at this stage.

Reply:
Hmm, oke. Still feels like this could and should be all merged and moved to a function in CoreBundle though...
However, when we add Sources to mongoDB that all might change 🤔

Reply:
For future reference: there is another function that should provide this functionality, but does not do that fully: CoreBundle->GatewayResourceService->findSourcesForUrl(). If that function has been updated, use that one and remove this one.